### ISSUE TITLE
Is this plugin still needed and if so how do I use it with sbt 1.8.2

### ISSUE THREAD

**Issue by oakstair** [sbt/sbt-maven-resolver#13]
Hi!

We are adding a **CodeArtifact** repository for all our artifacts.

Our maven projects resolves and published to CodeArtifact without any problems.

We currently are having great problems with our **Scala / sbt** based projects.

The current status is that we are able to resolve and publish but the artifacts in CodeArtifact ends up in an Unfinished state.

Then i found this link ...

https://stackoverflow.com/questions/64620633/publish-artifact-to-aws-codeartifact-with-sbt

... indicating that someone had the same problem 3 years ago and solved it by using this **sbt-maven-resolver** plugin.

But this plugin is not in maven central anymore and my question is if it is still relevant?

Thanks in advance!

**Comment by slandelle** [comment]
~~I recently experienced a similar issue with publishing on CodeArtifact where artifacts would be in Unpublished state. However, this only happened for new artifacts. Once an artifact was published manually, new versions would directly be published. I assumed this was how CodeArtifact works (I only publish with sbt there so didn’t compare with maven).
Is it was happens for you too?~~

My bad. We actually call the aws cli from our release GitHub Action to complete the publish.

**Comment by eed3si9n** [comment]
The plugin is available from https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.scala-sbt/sbt-maven-resolver/scala_2.12/sbt_1.0/0.1.0/.

**Comment by oakstair** [comment]
> The plugin is available from https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.scala-sbt/sbt-maven-resolver/scala_2.12/sbt_1.0/0.1.0/.

Yea but is it needed and will it solve my problem?



> I recently experienced a similar issue with publishing on CodeArtifact where artifacts would be in Unpublished state. However, this only happened for new artifacts. Once an artifact was published manually, new versions would directly be published. I assumed this was how CodeArtifact works (I only publish with sbt there so didn’t compare with maven). Is it was happens for you too?

Yes! works with maven but not with sbt. I will try your way to see if it works for new releases for artifcats that we manually has published.

**Comment by eed3si9n** [comment]
> Yea but is it needed and will it solve my problem?

I have no idea.

**Comment by oakstair** [comment]
@slandelle 
I just tested but we have to manually publish all new versions since they end up in this "Unfinished" state.

We have no plugin specified the only thing I can see that is related to publishing is 
```
    publishM2Configuration := publishM2Configuration.value.withOverwrite(true),

    publishTo := {
      val artifactoryUrl = "our-aws-repo-url"
      if (isSnapshot.value)
        Some("Snapshots" at s"${artifactoryUrl}")
      else
        Some("Releases" at s"${artifactoryUrl}")
    }

```

**Comment by oakstair** [comment]
Well it turned out that after adding this plugin, i.e.

`addSbtPlugin("org.scala-sbt" % "sbt-maven-resolver" % "0.1.0")`

we are able to publish Scala artifacts.

We do get som annoying info lines, one for each internal dependency ... so it seem like this is not 100% yeat :-)

`[info] Failed to read descriptor dependency: io.roaring#etl-cmn;1.1 {compile=[default(compile)]} from public, Failed to read artifact descriptor for io.roaring:etl-cmn:jar:1.1`

**Comment by shuch3ng** [comment]
> https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.scala-sbt/sbt-maven-resolver/scala_2.12/sbt_1.0/0.1.0/

I'm not able to access it. Is it just me, or is everyone experiencing the same issue?

<img width="929" alt="Image" src="https://github.com/user-attachments/assets/e4c31d3f-8c2f-4c0e-859e-773320293ab9" />

**Comment by shuch3ng** [comment]
> [@slandelle](https://github.com/slandelle) I just tested but we have to manually publish all new versions since they end up in this "Unfinished" state.
> 
> We have no plugin specified the only thing I can see that is related to publishing is
> 
> ```
>     publishM2Configuration := publishM2Configuration.value.withOverwrite(true),
> 
>     publishTo := {
>       val artifactoryUrl = "our-aws-repo-url"
>       if (isSnapshot.value)
>         Some("Snapshots" at s"${artifactoryUrl}")
>       else
>         Some("Releases" at s"${artifactoryUrl}")
>     }
> ```

No need to manually publish. I managed to include it within `sbt publish`.

```
  publish := {
    Def.sequential(
      publish,
      updateCodeArtifactStatus
    ).value
  },

  updateCodeArtifactStatus := {
    val ver      = version.value
    val scalaVer = scalaBinaryVersion.value
    val pkg      = s"${name.value}_$scalaVer"

    // see https://docs.aws.amazon.com/codeartifact/latest/ug/maven-curl.html Option 2
    val cmd = Seq(
      "aws",
      "codeartifact",
      "update-package-versions-status",
      "--domain",
      "my_domain",
      "--domain-owner",
      "111122223333",
      "--repository",
      "my_repo",
      "--format",
      "maven",
      "--namespace",
      organization.value,
      "--package",
      pkg,
      "--versions",
      ver,
      "--target-status",
      "Published"
    )

    val result = Process(cmd).!
    if (result != 0) {
      throw new MessageOnlyException(s"Failed to update CodeArtifact status for $pkg version $ver")
    }
  }
```

**Comment by mkurz** [comment]
The sbt team is running this repo, seems it is down, please follow:
- https://github.com/sbt/sbt/issues/8093

**Comment by slandelle** [comment]
@shuch3ng Yeah, this is what we do, using the AWS CLI, except we do it from our CI pipeline and not directly from sbt.

**Comment by mkurz** [comment]
Maybe it would make sense to finally publish the plugin to maven central: https://repo1.maven.org/maven2/org/scala-sbt/
@eed3si9n what do you think?

**Comment by eed3si9n** [comment]
Created a new issue for it. Closing this one.
